### PR TITLE
Initialise custom theme

### PR DESCRIPTION
**Pull request by clementbiron**
Initialize Open Terms Archive documentation custom theme that uses and adapts the styles used on the opentermsarchive.org website.

Changes were necessary to adapt the use of icons and react-use hooks on this stack.

image

### CODE REVIEW

File: themes/OpenTermsArchive/assets/js/header.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener('DOMContentLoaded', function () {
+	$header = document.querySelectorAll('.header')[0];
```
Comment (MattiSG) [inline themes/OpenTermsArchive/assets/js/header.js:2]: > This is equivalent to, with significantly lower performance costs:
```suggestion
	$header = document.querySelector('.header');
```

File: themes/OpenTermsArchive/assets/js/header.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+	$headerHandlers = document.querySelectorAll('.header_openLink, .header_closeLink');
+	$headerHandlers.forEach($el  => {
```
Comment (MattiSG) [inline themes/OpenTermsArchive/assets/js/header.js:4]: > This intermediate variable might not be necessary 🙂 
```suggestion
	document.querySelectorAll('.header_openLink, .header_closeLink').forEach($el  => {
```

**Review by MattiSG — CHANGES_REQUESTED** [review]
- The files `themes/OpenTermsArchive/assets/css/elements/buttons.css`, `themes/OpenTermsArchive/assets/css/elements/inlines.css`, `themes/OpenTermsArchive/layouts/_default/list.html` are empty, should they really be added? 🙂 
- Unless I am mistaken, this changeset should probably remove the previous theme.

File: .github/workflows/gh-pages.yml [hunk 3]
```diff
@@ -19,9 +19,10 @@ jobs:
         uses: peaceiris/actions-hugo@v2
         with:
           hugo-version: '0.107.0'
+          extended: true
```
[the remark this answers is not in the document]
Comment (clementbiron) [reply .github/workflows/gh-pages.yml:22]: According to this [build error](https://github.com/OpenTermsArchive/docs/actions/runs/3632108953/jobs/6127572113): `POSTCSS: failed to transform "css/loader.css" (text/css). Check your PostCSS installation; install with "npm install postcss-cli". See https://gohugo.io/hugo-pipes/postcss/: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information` and the [Hugo documentation](https://gohugo.io/troubleshooting/faq/#i-get-this-feature-is-not-available-in-your-current-hugo-version) we need to use Hugo extended. 

I made a comment 👍

File: content/_index.en.md [hunk 4]
```diff
@@ -1 +1,7 @@
-Welcome to the Open Terms Archive documentation!
+---
+type: "_default"
+layout: "single"
```
[the remark this answers is not in the document]
Comment (clementbiron) [reply content/_index.en.md]: You are right, it brings complexity where none is needed. I made a fixup to use the `index.html` layout.

**Comment by clementbiron** [comment]
> The files themes/OpenTermsArchive/assets/css/elements/buttons.css, themes/OpenTermsArchive/assets/css/elements/inlines.css, themes/OpenTermsArchive/layouts/_default/list.html are empty, should they really be added? 🙂

`buttons.css` and `inlines.css` are not necessary now but as I have started to import styles that are not yet in use I am also doing so with these.

`list.html` is created automatically by the `hugo new theme` command, this is also not necessary now, i deleted it.

File: themes/OpenTermsArchive/layouts/index.html [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{{ define "main" }}
```
Comment (MattiSG) [inline themes/OpenTermsArchive/layouts/index.html:1]: > This seems to be a duplicate of `themes/OpenTermsArchive/layouts/_default/single.html`

**Review by MattiSG — APPROVED** [review]

File: themes/OpenTermsArchive/layouts/index.html (hunk 5 above)
Comment (clementbiron) [reply themes/OpenTermsArchive/layouts/index.html:1]: Yes: `index.html` is the template layout for the homepage, `single.html` is the one for the pages that will be added and they actually use the same layout. Previously, in the frontmatter of the homepage, I had specified the layout `single`, but it is better to minimize the complexity of the frontmatter and have two identical files for now.

**Pull request merged**